src/main/kotlin/uk/gov/communities/prsdb/webapp/journeys/example/Destination.kt

```
@@ -0,0 +1,38 @@
+package uk.gov.communities.prsdb.webapp.journeys.example
+
+import org.springframework.web.servlet.ModelAndView
+import uk.gov.communities.prsdb.webapp.journeys.JourneyStep
+
+sealed class Destination {
+    fun toModelAndView(): ModelAndView =
+        when (this) {
+            is Step -> ModelAndView("redirect:${step.routeSegment}", mapOf("journeyId" to journeyId))
+            is ExternalUrl -> ModelAndView("redirect:$externalUrl", params)
+            is Template -> ModelAndView(templateName, content)
+        }
+
+    fun withContent(content: Map<String, Any?>): Destination =
+        when (this) {
+            is Template -> Template(templateName, this.content + content)
+            else -> this
```

Feels odd to let this function be called with some content but then silently do nothing?